sklearn/utils/estimator_checks.py

```diff
@@ -1191,6 +1192,58 @@ def check_estimators_pickle(name, estimator_orig):
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
 
+def check_pairwise_estimator_tag(name, estimator_orig):
+    attributes_to_check = [('metric', 'precomputed'),
+                           ('affinity', 'precomputed'),
+                           ('kernel', 'precomputed')]
+
+    # Check if _pairwise attribute is present - will be used later
+    has_pairwise_tag = (False
+                        if getattr(estimator_orig, '_pairwise', None) is None
```

Comment:
Why getattr and not hasattr?

Reply:
I think it's preferred to use `getattr` in place of `hasattr` since the latter simply tests the former for an `AttributeError`[1]. There's no benefit in terms of speed and maybe a little benefit in terms of code clarity, but there's the danger of a misrepresented `AttributeError`. In the interest of future-proofing (and going by general community consensus[2][3]) I think it may be better to adopt `getattr`.

References:
1. https://docs.python.org/3/library/functions.html#hasattr
2. https://stackoverflow.com/a/24971134/9366691
3. https://hynek.me/articles/hasattr/